src/main/java/com/commercetools/sync/commons/helpers/CustomReferenceResolver.java

```diff
@@ -107,7 +107,7 @@ protected CustomReferenceResolver(@Nonnull final S options, @Nonnull final TypeS
         try {
             final String customTypeKey = getKeyFromResourceIdentifier(custom.getType());
             return typeService.fetchCachedTypeId(customTypeKey);
-        } catch (ReferenceResolutionException exception) {
+        } catch (Exception exception) {
```

Comment:
this is not correct! it seems the code will swallow the other exceptions types as ReferenceResolutionException.

Reply:
This is the purpose. ReferenceResolutionException is considered as exception wrapper to indicate any exception thrown during resolveReference(), including any unchecked exception occurs in runtime.

Reply:
I wonder, why do we need to catch here at all? I see ReferenceResolutionException only as an additional wrapper which adds additional context/details to the error. Important is that such error should not break the whole batch processing but rather be passed to the error callback so that the user can react to it.

Reply:
I think, In line 109, the service class attempts to make query via jvm-sdk. If HttpException is suspected to occur, this coding line potentially involves. That's why I catch it here, or otherwise runtime exception is directly thrown to sync class without any handling taking place.

Reply:
The change has been rolled back